plugin/cassandra/plugin.go

```diff
@@ -0,0 +1,666 @@
+				Default:  "/var/vcap/packages/cassandra/bin",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_datadir",
+				Type:     "abspath",
+				Title:    "Path to Cassandra data/ directory",
+				Help:     "The absolute path to the data/ directory that contains the Cassandra database files.",
+				Default:  "/var/vcap/store/cassandra/data",
+				Required: true,
+			},
+			plugin.Field{
+				Mode:     "target",
+				Name:     "cassandra_tar",
+				Type:     "abspath",
+				Title:    "Path to `tar` utility",
+				Help:     "By default, the plugin will search the local `$PATH` to find the `tar` utility.",
+				Default:  DefaultTar,
+				Required: true,
+			},
+		},
+	}
+
+	plugin.Run(p)
+}
+
+type CassandraPlugin plugin.PluginInfo
+
+type CassandraInfo struct {
+	Host     string
+	Port     string
+	User     string
+	Password string
+	Keyspace string
+	BinDir   string
+	DataDir  string
+	Tar      string
+}
+
+// This function should be used to return the plugin's PluginInfo, however you decide to implement it
+func (p CassandraPlugin) Meta() plugin.PluginInfo {
+	return plugin.PluginInfo(p)
+}
+
+// Called to validate endpoints from the command line
+func (p CassandraPlugin) Validate(endpoint plugin.ShieldEndpoint) error {
+	var (
+		s    string
+		err  error
+		fail bool
+	)
+
+	s, err = endpoint.StringValueDefault("cassandra_host", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_host          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_host}          using default node @C{%s}\n", DefaultHost)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_host}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_port", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_port          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_port}          using default port @C{%s}\n", DefaultPort)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_port}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_user", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_user          %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_user}          using default user @C{%s}\n", DefaultUser)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_user}          @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValueDefault("cassandra_password", "")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_password      %s}\n", err)
+		fail = true
+	} else if s == "" {
+		fmt.Printf("@G{\u2713 cassandra_password}      using default password @C{%s}\n", DefaultPassword)
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_password}      @C{%s}\n", s)
+	}
+
+	s, err = endpoint.StringValue("cassandra_keyspace")
+	if err != nil {
+		fmt.Printf("@R{\u2717 cassandra_keyspace      %s}\n", err)
+		fail = true
+	} else {
+		fmt.Printf("@G{\u2713 cassandra_keyspace}      @C{%s}\n", s)
```

Comment:
Is there any particular format to the keyspace argument?  The documentation comments above imply that all keyspace names should start with `ks` -- if so, we should validate that here.

Reply:
The `ks` prefix is a convention [enforced by the broker](https://github.com/orange-cloudfoundry/cassandra-cf-service-boshrelease/blob/master/src/cassandra-open-service-broker/src/main/java/com/orange/oss/osb/cassandra/util/Converter.java#L8) in the Cassandra BOSH Release we are currently building. So there's nothing to validate here. The `ksXXX` example is just a suggestion.

Reply:
Carry on then 😃